inst/include/dust/densities.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef DUST_DENSITIES_HPP
+#define DUST_DENSITIES_HPP
+
+#include <cmath>
+
+namespace dust {
+
+template <typename T>
+T maybe_log(T x, bool log) {
+  return log ? x : std::exp(x);
+}
+
+template <typename T>
+T lchoose(T n, T k) {
+  return std::lgamma(static_cast<T>(n + 1)) -
+    std::lgamma(static_cast<T>(k + 1)) -
+    std::lgamma(static_cast<T>(n - k + 1));
+}
+
+template <typename T>
+T lfactorial(T x) {
+  return std::lgamma(x + 1);
+}
+
+template <typename T>
+T lbeta(T x, T y) {
+  return lgamma(x) + lgamma(y) - lgamma(x + y);
+}
```

Comment:
I can't remember if we're C++14 or C++17, but if the latter this (unlogged) is in <cmath> as std::beta

Reply:
We really need the logged version though, and that's not coming in either standard